internal/gengapic/testdata/method_GetAnotherThing.want

```diff
@@ -0,0 +1,55 @@
+func (c *fooGRPCClient) GetAnotherThing(ctx context.Context, req *mypackagepb.InputType, opts ...gax.CallOption) (*mypackagepb.OutputType, error) {
+	if _, ok := ctx.Deadline(); !ok && !c.disableDeadlines {
+		cctx, cancel := context.WithTimeout(ctx, 10000 * time.Millisecond)
+		defer cancel()
+		ctx = cctx
+	}
+	routingHeaders := ""
+	routingHeadersMap := make(map[string]string)
+	routingRegexp0 := regexp.MustCompile("(.*)")
+	headerName0 := "other"
+	headerValue0 := ""
+	if routingRegexp0.MatchString(req.GetOther()){headerValue0 = url.QueryEscape(routingRegexp0.FindStringSubmatch(req.GetOther())[1])}
+	if len(headerValue0) > 0{routingHeadersMap[headerName0] = headerValue0}
+	routingRegexp1 := regexp.MustCompile("(?P<name>projects/[^/]+)/foos")
+	headerName1 := "name"
+	headerValue1 := ""
+	if routingRegexp1.MatchString(req.GetOther()){headerValue1 = url.QueryEscape(routingRegexp1.FindStringSubmatch(req.GetOther())[1])}
+	if len(headerValue1) > 0{routingHeadersMap[headerName1] = headerValue1}
+	routingRegexp2 := regexp.MustCompile("(?P<foo_name>projects/[^/]+)/bars/[^/]+(?:/.*)?")
+	headerName2 := "foo_name"
+	headerValue2 := ""
+	if routingRegexp2.MatchString(req.GetAnother()){headerValue2 = url.QueryEscape(routingRegexp2.FindStringSubmatch(req.GetAnother())[1])}
+	if len(headerValue2) > 0{routingHeadersMap[headerName2] = headerValue2}
+	routingRegexp3 := regexp.MustCompile("(?P<foo_name>projects/[^/]+/foos/[^/]+)/bars/[^/]+(?:/.*)?")
+	headerName3 := "foo_name"
+	headerValue3 := ""
+	if routingRegexp3.MatchString(req.GetAnother()){headerValue3 = url.QueryEscape(routingRegexp3.FindStringSubmatch(req.GetAnother())[1])}
+	if len(headerValue3) > 0{routingHeadersMap[headerName3] = headerValue3}
+	routingRegexp4 := regexp.MustCompile("(?P<nested_name>.*)")
```

Comment:
```suggestion
	if reg := regexp.MustCompile("(?P<foo_name>projects/[^/]+)/bars/[^/]+(?:/.*)?"); reg.MatchString(req.GetAnother()) {
	    routingHeadersMap["foo_name"] = url.QueryEscape(reg.FindStringSubmatch(req.GetAnother())[1])
	} else if reg = regexp.MustCompile("(?P<foo_name>projects/[^/]+/foos/[^/]+)/bars/[^/]+(?:/.*)?"); reg.MatchString(req.GetAnother()) {
	    routingHeadersMap["foo_name"] = url.QueryEscape(reg.FindStringSubmatch(req.GetAnother())[1])
	}
```

Here is another idea of how we can structure the generated code taking advantage of Go's ability to define local variables in an `if` and avoiding unnecessary creation of variables for things that the generator already knows about i.e. the header name (we can just use the raw value). I think this greatly improves readability.

Reply:
that's really neat; thank you for explaining! I've refactored `insertDynamicRequestHeaders` to match the above.